src/common/Utilities/Optional.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
```

Comment:
is this taken from tc or mangos?

Reply:
I just copy/paste the first AC copyright header I see everywhere, it's from TC tho.

Shall I remove the MaNGOS one or leave it as-is?

Reply:
if it's from TC (and not from MaNGOS) you can remove the MaNGOS line.

Just FYI: for new files in AC (that are not imported from TC/cmangos) we have another header